### ISSUE TITLE
Wrong behavior in 'run code' and 'submit'- 202. Happy Number

### ISSUE THREAD

**Issue by Killea** [LeetCode-Feedback/LeetCode-Feedback#8690]
#### Your LeetCode username
https://leetcode.com/Hanking/   Hanking

#### Category of the bug
- [ ] Question
- [ ] Solution
- [ *] Language
- [ ] Missing Test Cases 

#### Description of the bug

Different behaviors in 'Run Code' and 'Submit'. 
'Run code' gets **true**; 'Submit' gets '**false**'

#### Code you used for Submit/Run operation

// JS or TS
```
let cnt = 0 
 var isHappy = function(n) {
    cnt ++;
    if (cnt > 1000)
    {
        return false;
    }
    const numbers = [...String(n)]
    let sum =0;
    for(let i=0; i<numbers.length; i++)
        {
            sum += Number(numbers[i]) * Number(numbers[i]);
        }
    if (sum === 1)
        {
            return true
        }
    else if (sum >1 && sum <4 || sum === 0)
        {
            return false
        }
    else {
        return isHappy(Number(sum))
    }
};
```

#### Expected behavior

Number 1 should get **true** as 'Run code does'.
In short, 'run code' and 'submit' should have the same behavior.

#### Screenshots

![image](https://user-images.githubusercontent.com/5551824/185021056-84d447f4-595a-4cfc-9d05-8650d1821d35.png)
![image](https://user-images.githubusercontent.com/5551824/185021113-bdce54a8-9cac-46e6-9962-7c97d9cab272.png)

**Comment by LC-Pam** [comment]
Thank you for reaching out to us! The reasons and solutions for different "outputs" from "Run Code" and "Submit" are well documented in this article: https://support.leetcode.com/hc/en-us/articles/360011834174-I-encountered-Wrong-Answer-Runtime-Error-for-a-specific-test-case-When-I-test-my-code-using-this-test-case-it-produced-the-correct-output-Why-. Please check it out.

We appreciate your time and support! If you have any other questions, please don't hesitate to let us know.
